mkdocs/livereload/__init__.py

```diff
@@ -86,6 +86,21 @@ def callback(event):
         handler.on_any_event = callback
         self.observer.schedule(handler, path, recursive=recursive)
 
+        def schedule(handler, path, recursive=True):
+            if schedule.depth >= schedule.limit:
+                return
+            schedule.depth += 1
+            if os.path.islink(path):
+                self.observer.schedule(handler, os.path.realpath(path), recursive=recursive)
+            elif recursive and os.path.isdir(path):
+                for name in os.listdir(path):
+                    schedule(handler, os.path.join(path, name), recursive=recursive)
+            schedule.depth -= 1
+
+        schedule.limit = 9
+        schedule.depth = 1
```

Comment:
**Untested code**, and just for your reference, because I don't think this should be the final version.

A simpler way to write the same:

```suggestion
        def schedule(handler, path, limit=9):
            if os.path.islink(path):
                self.observer.schedule(handler, os.path.realpath(path), recursive=recursive)
            elif recursive and os.path.isdir(path) and limit > 0:
                for name in os.listdir(path):
                    schedule(handler, os.path.join(path, name), limit - 1)

```

Reply:
Yes I tried that as well (in a similar variant passing the limit / depth counter as argument), however it changes the signature compared to the parenting schedule implementation.

functionality wise it works, can confirm that as I tried it out already earlier.

Reply:
> however it changes the signature

It does indeed change the signature of the function (the args it accepts, if that's what you're referring to), but that doesn't matter because the function is fully internal (impossible for a user to call).

---

And a comment that I add since you mention wanting to improve Python knowledge.

`some_function.some_var = 2` is usually seen as "obscure". The main argument against using this ability is that most readers would be surprised that it exists, or generally not expect it, meaning it might take away from their process of understanding the code.

The direct alternative here is plain variables but with the `nonlocal` keyword. But in this case I still think that's moot because I think my suggestion removes that need entirely.